web/docs/auth/migrate-from-11.md

```diff
@@ -0,0 +1,338 @@
+
+If you have an existing app with authentication set up, you can follow these steps to migrate to the new auth system:
+
+1. Upgrade Wasp to the latest 0.12 version.
+1. Bump the version to `^0.12.0` in `main.wasp`.
+1. Create the new auth tables in the database by running:
+
+  ```bash
+  wasp db migrate-dev
+  ```
+
+  You should see the new `Auth`, `AuthIdentity` and `Session` tables in your database.
+
+1. Create a data migration script in `src/server/migrateToNewAuth.ts`
+
+    - Below you can find [examples of migration scripts](#example-migration-scripts) for each of the auth methods. They should be fine to use as-is, but you can also modify them to your needs.
+
+1. Add the migration script(s) to the `db.seeds` config:
+  ```wasp title="main.wasp"
+  app myApp {
+    wasp: {
+      version: "^0.12.0"
+    },
+    // ...
+    db: {
+      seeds: [
+        import { migrateEmailAuth } from "@server/migrateToNewAuth.js",
+        import { migrateGoogleAuth } from "@server/migrateToNewAuth.js",
+      ]
+    },
+  }
+  ```
+1. Run the migration script(s) by running:
+  ```bash
+  wasp db seed
+  ```
+  If you have multiple migration scripts, you can pick which one to run by selecting it from the list.
+
+  :::info Migrating a deployed app
+
+    _First, deploy your app_ with the changes from previous steps. Then you'll need to run the migration script on the production database with `wasp db seed` command.
+
+    **Running the migration script on the production database**
+
+    We wrote instructions on how to do it for **Fly.io** deployments here: https://github.com/wasp-lang/wasp/issues/1464 
+
+    The instructions should be similar for other deployment providers: setting up some sort of an SSH tunnel from your local machine to the production database and running the migration script locally with `DATABASE_URL` pointing to the production database.
+  :::
+
+1. Migrate to using the new auth helper functions for getting the `email` or `username` of the currently logged-in user. Read more about the helpers in the [Auth Entities](./entities#accessing-the-auth-fields) section.
+
+  The helpers you'll probably use are the `getEmail` and `getUsername` helpers.
+
+  :::info Migrating a deployed app
+
+    Now, _deploy the app again_ with the new auth helpers in place.
+
+  :::
+
+1. Verify that the auth still works by logging in with each of the auth methods.
+
+### 2. Cleanup the Old Auth System
+
+Now we need to clean up the old auth system:
+
+1. Delete auth-related fields from `User` entity.
+1. Remove the `externalAuthEntity` from the `auth` config and the `SocialLogin` entity (if you used it).
+1. Run `wasp db migrate-dev` again to remove the redundant fields from the database.
+1. You can now delete the migration script and the `db.seeds` config.
+
+:::info Migrating a deployed app
+
+  One last time, _deploy the app_ with the old auth system cleaned up. Your app should now be migrated to the new auth system.
```

Comment:
Mention that this deployment, if they are using our Dockerfile for server, will run the new db migrations for them and therefore delete the old data from the db (which is ok, because we migrated it before).